### PR TITLE
Winbuild: Updated BUILD.WINDOWS.txt to reflect more recent data

### PR DESCRIPTION
**Pull request by kdekker**
1. Use URLs that point to newer versions of SDKs/documentation
2. Updated text to reflect more recent Visual Studio versions (2015 and 2017)

### CODE REVIEW

**Comment by bagder** [comment]
This PR has 11 irrelevant commits and 1 that you want merged...

**Comment by bagder** [comment]
I cherry-picked the single patch and merged that now. Please also check our template for commit messages, as that makes it even easier for us to receive your changes.

Thanks!

**Comment by kdekker** [comment]
I reverted all on master branch of my own clone + forced (based on the comments of Marcel Raad) a sync with the remote repository. I don't know how to find your template. I'm almost struggling for 3 days how to create a PR. I understand that you have not that much time to explain, but just 'check our template' causes me to spend very much time, on just a simple documentation change. It requires really much courage (and time) to contribute to curl... I'm a little bit disappointed how much.

**Comment by bagder** [comment]
> It requires really much courage (and time) to contribute to curl... I'm a little bit disappointed how much.

To me, it seems this time you speak of is mostly you not knowing the primary tool we use: git. I don't know what we can do to make that ride easier... Also, making PRs on github is widely documented on github and elsewhere and also a procedure that is shared by millions of projects so lots of users already know how, plus once you've learned you'll be able to use this knowledge in many other projects.

The preferred commit message style is described in the regular [contribute page](https://curl.haxx.se/dev/contribute.html). See the "Write good commit messages" subtitle.

File: winbuild/BUILD.WINDOWS.txt [hunk 1]
```diff
@@ -12,7 +12,7 @@ Building with Visual C++, prerequisites
 
    The latest Platform SDK can be downloaded freely from:
 
-    https://msdn.microsoft.com/en-us/windows/bb980924
+    https://developer.microsoft.com/nl-nl/windows/downloads/sdk-archive
```
Comment (MarcelRaad) [inline winbuild/BUILD.WINDOWS.txt]: `en-us` would probably be understood by more people than `nl-nl` ;-)

**Comment by kdekker** [comment]
You are right. Unfortunately, the MS website is doing this 'automagically'. 
But how to continue now? As advised, I dropped the original fork and created a new one. 

If this change was already committed on the master branch of the cURL repository, I will create a new PR and fix this URL. If this change was not yet committed, I will create a new branch in my own fork, and also create a new PR.  If I understood Daniel's comments, this change is as #73070e8 on the master? 

But as I made already some other mistakes, I really like to get a confirmation.

**Comment by MarcelRaad** [comment]
Ah, you're right, this is already in master. You will get it by doing a `git pull --ff-only` on your master branch. After that, you can create a new branch from that. Thanks!

**Comment by kdekker** [comment]
Fixed now by filing a new PR (see above).

**Pull request closed without merging**